### PR TITLE
ref(ui): Use DropdownMenuV2 in Discover cards

### PR DESCRIPTION
**Pull request by vuluongj20**
Before:
Screen Shot 2022-02-04 at 12 56 54 PM

After:
Screen Shot 2022-02-04 at 12 57 14 PM

### CODE REVIEW

File: static/app/views/eventsV2/queryList.tsx [hunk 1]
```diff
@@ -162,6 +150,33 @@ class QueryList extends React.Component<Props> {
     return cards;
   }
 
+  renderDropdownMenu(items) {
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/queryList.tsx:153]:
```suggestion
  renderDropdownMenu(items: MenuItemProps[]) {
```

File: tests/js/sentry-test/dropdownMenu.tsx [hunk 2]
```diff
@@ -36,8 +36,11 @@ type SelectDropdownItemProps = {
 };
 
 /**
- * Select a dropdown menu item. This function works for both top-level and
- * nested items.
+ * @deprecated
+ * (This function is only necessary for enzyme tests, which we are migrating
```
Comment (priscilawebdev) [inline tests/js/sentry-test/dropdownMenu.tsx:40]: 💜

File: static/app/views/eventsV2/queryList.tsx [hunk 3]
```diff
@@ -262,6 +275,32 @@ class QueryList extends React.Component<Props> {
       const dateStatus = <TimeSince date={savedQuery.dateUpdated} />;
       const referrer = `api.discover.${eventView.getDisplayMode()}-chart`;
 
+      const menuItems = canAddToDashboard => [
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/queryList.tsx:278]:
```suggestion
       const menuItems = (canAddToDashboard: boolean) => [
```

**Comment by priscilawebdev** [comment]
@vuluongj20 the dropdownMenuV2 looks great, but for some reason, I keep seeing this scrollbar on Safari (all viewports) and Chrome (large viewports)

drop-down-scrollbar
)

**Comment by vuluongj20** [comment]
> @vuluongj20 the dropdownMenuV2 looks great, but for some reason, I keep seeing a scrollbar on Safari (all viewports) and Chrome/Firefox (large viewports)
> 
> drop-down-scrollbar

@priscilawebdev ahh the video was really helpful! Figured out what happened (hopefully). In `DropdownMenuV2`, I set the overflow behavior to `overflow-y: scroll`. This means that the dropdown menu will always have scroll, even if there is no overflow. Because you seem to have chosen to always show scrollbars (in your system preferences), you will see an empty scrollbar track without any scroll thumb. I didn't change that setting, so I didn't see the bar. 

I set the overflow behavior back to `overflow-y: auto` (https://github.com/getsentry/sentry/pull/31616/commits/a589e8f927992e9bb8e2af39465279af6fbc139a) and it seems to behave as expected now. Tell me if this has been resolved!

**Review by priscilawebdev — APPROVED** [review]

**Comment by priscilawebdev** [comment]
It works now! Thanks, @vuluongj20 🙏 

> Because you seem to have chosen to always show scrollbars (in your system preferences)

I had to fix a similar issue in the past and left the scrollbar option on in my settings 🙂

**Pull request merged**